src/main/kotlin/com/exactpro/th2/rptdataprovider/Extensions.kt

```diff
@@ -58,6 +61,48 @@ suspend fun <T> logTime(methodName: String, lambda: suspend () -> T): T? {
     }
 }
 
+
+suspend fun Writer.eventWrite(event: SseEvent) {
+    withContext(Dispatchers.Default) {
+        if (event.event != null) {
+            write("event: ${event.event}\n")
+        }
+
+        for (dataLine in event.data.lines()) {
+            write("data: $dataLine\n")
+        }
+
+        if (event.id != null) {
+            write("id: ${event.id}\n")
+        }
+        write("\n")
+        flush()
+    }
+}
+
+suspend fun Writer.asyncClose() {
+    withContext(Dispatchers.IO) {
```

Comment:
Dispatchers.IO may use more that one thread. Are we sure writer is ok with this?